site/oauth2-examples-azure.md

```diff
@@ -0,0 +1,237 @@
+
+Once you've selected users and groups, click on the **Select** button.
+
+Back to the **Add assignment** pane, below **Select a Role**, click on *None Selected* and, on the **Select a role** pane that has just opened on the right, search and select the role you want to assign to the selected users.
+
+<g-emoji class="g-emoji" alias="bulb" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4a1.png">💡</g-emoji> If only one role is available for your application, it would be automatically selected and greyed by default
+
+Choose a role (only a single role can be selected at a time), click on the **Select** button, and click on the **Assign** button to finalize the assignment of users and groups to the app.
+
+Repeat the operations for all the roles you want to assign.
+
+## Configure RabbitMQ to use Azure AD as OAuth 2.0 authentication backend
+
+The configuration on Azure side is done. You now have to configure RabbitMQ to use the resources you just created.
+
+[rabbitmq.config](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/master/conf/azure/rabbitmq.config) is a sample RabbitMQ advanced configuration to **enable Azure AD as OAuth 2.0 authentication backend** for the RabbitMQ Management UI.
+
+Update it with the following values (you should have noted these in the previous steps):
+* **Tenant ID** associated to the app that we registered in Azure AD
+* **Application ID** associated to the app that we registered in Azure AD
+* Value of the **secret** we created for our app in Azure AD
+* Value of the **jwks_uri** key from `https://login.microsoftonline.com/{TENANT_ID}/v2.0/.well-known/openid-configuration`
+
+<pre class="lang-bash">
+$ vi rabbitmq.config
+[
+  {rabbit, [
+   {auth_backends, [rabbit_auth_backend_oauth2, rabbit_auth_backend_internal]}
+  ]},
+  {rabbitmq_management, [
+     {oauth_enabled, true},
+     {oauth_client_id, "PUT YOUR AZURE AD APPLICATION ID"},
+     {oauth_client_secret, "PUT YOUR AZURE AD APPLICATION SECRET"},
+     {oauth_provider_url, "https://login.microsoftonline.com/AZURE_AD_TENANT_ID"}
+ ]},
+ {rabbitmq_auth_backend_oauth2, [
+   {resource_server_id, &lt;&lt;"PUT YOUR AZURE AD APPLICATION ID"&gt;&gt;},
+   {extra_scopes_source, &lt;&lt;"roles"&gt;&gt;},
+   {key_config, [
+     {jwks_url, &lt;&lt;"PUT YOUR AZURE AD JWKS URI VALUE"&gt;&gt;}
+   ]}
+ ]}
+].
+</pre>
+
+> :warning: Please update the file available in this tutorial ([here](../conf/azure/rabbitmq.config)), as it will be automatically loaded in the RabbitMQ instance that we are going to deploy later in this tutorial
+
+### Generate SSL certificate and key
+> :warning: Remember when you have registered your app on Azure AD that it only allows **https** protocol for OAuth2 **Redirect URI**? We will thus need to enable HTTPS for RabbitMQ Management UI amd its underlying API.
+
```

Comment:
I know in VMWare documentation, the use of "Warning" in documentation is not permitted - see below - I suggest using "Important" here instead - maybe put it in caps if you want to draw more attention to it

Warning | Do not use. VMware software documentation does not require this level of notice. Warning notices are typically used in hardware documentation to alert the reader to a potentially hazardous situation that could result in serious injury or death.


-- | --